src/chat/chatReducers.js

```diff
@@ -104,6 +104,26 @@ export default (state: ChatState = initialState, action: Action) => {
           return msg;
         }, {}),
       };
+      if (!stateChange) {
+        // new message is in new narrow in which we don't have any message
+        const { message } = action;
+        let key;
```

Comment:
Instead of these `ifs` use the `getNarrowFromMessage` function.

Reply:
yup! I missed it before.